code/modules/mob/living/carbon/xenomorph/castes/baneling/abilities_baneling.dm

```diff
@@ -0,0 +1,118 @@
+// ***************************************
+// *********** Baneling Explode
+// ***************************************
+/datum/action/xeno_action/baneling_explode
+	name = "Baneling Explode"
+	action_icon_state = "baneling_explode"
+	desc = "Explode and spread dangerous toxins to kill your foes"
+	ability_name = "baneling explode"
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_BANELING_EXPLODE,
+	)
+
+/datum/action/xeno_action/baneling_explode/give_action(mob/living/L)
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = L
+	RegisterSignal(X, COMSIG_MOB_DEATH, PROC_REF(handle_smoke))
+
+/datum/action/xeno_action/baneling_explode/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
+	handle_smoke(ability = TRUE)
+	X.death(FALSE)
+
+/datum/action/xeno_action/baneling_explode/proc/handle_smoke(mob/M, ability = FALSE)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/X = owner
+	var/turf/owner_T = get_turf(X)
+	var/datum/effect_system/smoke_spread/smoke
+	switch(X.selected_reagent)
+		if(/datum/reagent/toxin/xeno_neurotoxin)
+			smoke = new /datum/effect_system/smoke_spread/xeno/neuro/medium(owner_T)
```

Comment:
why not just use an assoc list for this?

Reply:
I did but QV wanted me to make this child of defilers reagent selection